src/hooks/useScrollShadow.js

```diff
@@ -0,0 +1,94 @@
+// very difficult to test with JSDom, some basic interaction is tested in ScrollableContainer
+/* istanbul ignore file */
+import { useEffect } from 'react'
+import throttle from 'lodash.throttle'
+
+const BOX_SHADOW_INITIAL = '0 0 0 1px rgba(0, 0, 0, 0.1)'
+const BOX_SHADOW_SCROLLED =
+  '0 0 0 1px rgba(0, 0, 0, 0.1), 0 0 0 5px rgba(0, 0, 0, 0.05)'
+
+export default function useScrollShadow({
+  bottomRef,
+  scrollableRef,
+  topRef,
+  shadows = {},
+}) {
+  const initialShadow = shadows.initial || BOX_SHADOW_INITIAL
+  const scrolledShadow = shadows.scrolled || BOX_SHADOW_SCROLLED
+
+  useEffect(() => {
+    setTimeout(() => {
```

Comment:
should we clear timeout when effect re-runs? Or rather when it's unmounted, by using the returned function from the `useEffect`?

Reply:
👍